kie-wb-common-react/boxed-expression-editor/src/components/LiteralExpression/LiteralExpression.tsx

```diff
@@ -16,33 +16,46 @@
 
 import "./LiteralExpression.css";
 import * as React from "react";
-import { useCallback, useState } from "react";
-import { ExpressionProps, LiteralExpressionProps } from "../../api";
+import { useCallback, useEffect, useState } from "react";
+import { ExpressionProps, LiteralExpressionProps, LogicType } from "../../api";
 import { TextArea } from "@patternfly/react-core";
 import { EditExpressionMenu } from "../EditExpressionMenu";
 
 export const LiteralExpression: React.FunctionComponent<LiteralExpressionProps> = ({
   content,
   dataType,
   name,
-  updateNameAndDataTypeCallback,
+  onUpdatingNameAndDataType,
+  isHeadless = false,
 }: LiteralExpressionProps) => {
   const [expressionName, setExpressionName] = useState(name);
   const [expressionDataType, setExpressionDataType] = useState(dataType);
   const [literalExpressionContent, setLiteralExpressionContent] = useState(content);
 
+  useEffect(() => {
+    if (window.beeApi?.broadcastLiteralExpressionDefinition) {
+      window.beeApi.broadcastLiteralExpressionDefinition({
+        name: expressionName,
+        dataType: expressionDataType,
+        logicType: LogicType.LiteralExpression,
+        content: literalExpressionContent,
+      });
+    }
+  });
+
```

Comment:
In this way this effect is running on each render. Is that the intended behavior?
If its purpose is to propagate changes to an upper level then it's probably not a great idea because it could run when there are no actual changes.
It could be invoked when changing the expression content or its name/data type. But maybe I got this wrong!

Reply:
In this specific case, re-renders happen only when name, dataType and content change, so it is working fine. But, in general I will apply your suggestion, to be performance-wise :+1: